pppd/plugins/radius/util.c

```diff
@@ -78,7 +78,7 @@ rc_mksid (void)
   snprintf(buf, sizeof(buf), "%08lX%04X%02hX",
 	   (unsigned long int) time (NULL),
 	   (unsigned int) getpid (),
-	   cnt & 0xFF);
+	   (unsigned short) (cnt & 0xFF));
```

Comment:
This really seems like the compiler is being over-zealous. By C argument passing rules, this is going to get coerced back to int anyway. And it is clear that the value fits in 16 bits, so there is no possible loss of information in the hX conversion.